_inc/client/components/jetpack-notices/index.jsx

```diff
@@ -98,53 +101,43 @@ export class DevModeNotice extends React.Component {
 				reasons = [];
 
 			if ( devMode.filter ) {
-				reasons.push(
-					__( '{{li}}The jetpack_development_mode filter is active{{/li}}', {
-						components: {
-							li: <li />,
-						},
-					} )
-				);
+				reasons.push( __( 'The jetpack_development_mode filter is active', 'jetpack' ) );
 			}
 			if ( devMode.constant ) {
-				reasons.push(
-					__( '{{li}}The JETPACK_DEV_DEBUG constant is defined{{/li}}', {
-						components: {
-							li: <li />,
-						},
-					} )
-				);
+				reasons.push( __( 'The JETPACK_DEV_DEBUG constant is defined', 'jetpack' ) );
 			}
 			if ( devMode.url ) {
-				reasons.push(
-					__( '{{li}}Your site URL lacks a dot (e.g. http://localhost){{/li}}', {
-						components: {
-							li: <li />,
-						},
-					} )
-				);
+				reasons.push( __( 'Your site URL lacks a dot (e.g. http://localhost)', 'jetpack' ) );
 			}
 
-			const text = __(
-				'Currently in {{a}}Development Mode{{/a}} (some features are disabled) because: {{reasons/}}',
+			const text = createInterpolateElement(
+				/* translators: reasons is an unordered list of reasons why a site may be in Development mode. */
+				__(
+					'Currently in <a>Development Mode</a> (some features are disabled) because: <reasons/>',
+					'jetpack'
+				),
 				{
-					components: {
-						a: (
-							<a
-								href={ getRedirectUrl( 'jetpack-support-development-mode' ) }
-								target="_blank"
-								rel="noopener noreferrer"
-							/>
-						),
-						reasons: <ul>{ reasons }</ul>,
-					},
+					a: (
+						<a
+							href={ getRedirectUrl( 'jetpack-support-development-mode' ) }
+							target="_blank"
+							rel="noopener noreferrer"
+						/>
+					),
+					reasons: (
+						<ul>
+							{ reasons.map( ( reason, i ) => {
+								return <li key={ i }>{ reason }</li>;
+							} ) }
+						</ul>
+					),
```

Comment:
I changed the way this was built a bit; I did not see the point in having the `<li>` tags inside the translations. You can test this on a site with development mode active.

<img width="833" alt="screenshot 2020-07-15 at 18 36 26" src="https://user-images.githubusercontent.com/426388/87636812-95e78080-c741-11ea-9ead-2b09f3459816.png">